### PR TITLE
[CE-4220] Fixed a couple issues related to cancelling subscription long polling requests

### PR DESCRIPTION
**Pull request by CraigLn**

### CODE REVIEW

File: Sources/PubNub/Subscription/SubscribeSessionFactory.swift [hunk 1]
```diff
@@ -66,7 +67,7 @@ public class SubscribeSessionFactory {
 
     PubNub.log.debug("Creating new session for with hash value \(config.subscriptionHashValue)")
     return sessions.lockedWrite { dictionary in
-      let sessionReplaceable = session ?? HTTPSession(configuration: URLSessionConfiguration.subscription)
+      let sessionReplaceable = session ?? HTTPSession(configuration: URLSessionConfiguration.subscription, sessionQueue: subscribeQueue)
```
Comment (madisonsmith) [inline Sources/PubNub/Subscription/SubscribeSessionFactory.swift]: Codacy Issue found: [Line should be 120 characters or less: currently 136 characters](https://app.codacy.com/app/PubNub/swift/pullRequest?prid=4633870)

**Comment by madisonsmith** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/PubNub/swift/pullRequest?prid=4633870&bid=15529310)

File: Tests/PubNubTests/Integration/SubscriptionIntegrationTests.swift [hunk 2]
```diff
@@ -98,17 +103,22 @@ class SubscriptionIntegrationTests: XCTestCase {
           XCTAssertFalse(pubnub.subscribedChannels.contains(self.testChannel))
           unsubscribeExpect.fulfill()
         }
+      case .messageReceived:
+//        print("messageReceived: \(message)")
+        pubnub.unsubscribe(from: [self.testChannel])
+        publishExpect.fulfill()
       case let .connectionStatusChanged(status):
-        print("Status: \(status)")
+//        print("connectionStatusChanged: \(status)")
```
Comment (davidnub) [inline Tests/PubNubTests/Integration/SubscriptionIntegrationTests.swift]: Do we leave these commented out lines in for future/later usage?

Comment (CraigLn) [reply Tests/PubNubTests/Integration/SubscriptionIntegrationTests.swift]: Yeah it's just a convenience for testing.

**Pull request merged**